llvm/lib/Target/AMDGPU/SIInsertWaitcnts.cpp

```diff
@@ -1708,6 +1710,19 @@ bool SIInsertWaitcnts::insertWaitcntInBlock(MachineFunction &MF,
     }
 
     ++Iter;
+    if (ST->isPreciseMemoryEnabled() && Inst.mayLoadOrStore()) {
+      auto Builder =
+          BuildMI(Block, Iter, DebugLoc(), TII->get(AMDGPU::S_WAITCNT))
+              .addImm(0);
+      if (IsGFX10Plus) {
+        Builder =
+            BuildMI(Block, Iter, DebugLoc(), TII->get(AMDGPU::S_WAITCNT_VSCNT))
+                .addReg(AMDGPU::SGPR_NULL, RegState::Undef)
+                .addImm(0);
+      }
+      OldWaitcntInstr = Builder.getInstr();
```

Comment:
Nit: if you're going to set OldWaitcntInstr then really it ought to point to the first in a sequence of waitcnts, not the last.

Reply:
Done.